mmedit/datasets/pipelines/blur_kernels.py

```diff
@@ -0,0 +1,530 @@
+# This code is referenced from BasicSR with modifications.
+# Reference: https://github.com/xinntao/BasicSR/blob/master/basicsr/data/degradations.py  # noqa
+# Original licence: Copyright (c) 2020 xinntao, under the Apache 2.0 license.
+import random
+
+import numpy as np
+from scipy import special
+
+
+def get_rotated_sigma_matrix(sig_x, sig_y, theta):
+    """Calculate the rotated sigma matrix (two dimensional matrix).
+
+    Args:
+        sig_x (float): Standard deviation along the horizontal direction.
+        sig_y (float): Standard deviation along the vertical direction.
+        theta (float): Rotation in radian.
+
+    Returns:
+        ndarray: Rotated sigma matrix.
+    """
+
+    diag = np.array([[sig_x**2, 0], [0, sig_y**2]])
+    rot = np.array([[np.cos(theta), -np.sin(theta)],
+                    [np.sin(theta), np.cos(theta)]])
+
+    return np.matmul(rot, np.matmul(diag, rot.T))
```

Comment:
this mat will be float64.

Reply:
A general comment: operations in this file seem to be all float64. But it is slower than float32. Our convention is to use float32 for pipelines if possible. Float64 is used only when the precision is critical for that op.

One simple way to fix this is to add a line of note in the docstring that this op is computed in double precision.
A better fix is to check if some of them are valid in float32, and use float32 for those

![image](https://user-images.githubusercontent.com/9464825/132610813-c8b1f630-abe3-4a16-afd8-37de14a9d675.png)